apsuite/optics_analysis/orbit_correction.py

```diff
@@ -0,0 +1,164 @@
+        self.cv_idx = self.respm.fam_data['CV']['index']
+        self.corr_idx = np.vstack((self.ch_idx, self.cv_idx))
+
+    @property
+    def _nbpm(self):
+        return len(self.bpm_idx)
+
+    @property
+    def _nch(self):
+        return len(self.ch_idx)
+
+    @property
+    def _ncv(self):
+        return len(self.cv_idx)
+
+    @property
+    def _ncorr(self):
+        return self._nch + self._ncv
+
+    def calc_jacobian_matrix(self, model=None):
+        """."""
+        if model is None:
+            model = self.model
+        self.respm.model = model
+        return self.respm.get_respm()
+
+    def _get_orbit_residue(self, model):
+        if self.dim == '4d':
+            cod = pyaccel.tracking.find_orbit4(model, indices='open')
+        elif self.dim == '6d':
+            cod = pyaccel.tracking.find_orbit6(model, indices='open')
+        codx = cod[0, self.bpm_idx].flatten()
+        cody = cod[2, self.bpm_idx].flatten()
+        res = np.hstack((codx, cody))
+        return res
+
+    def get_kicks(self, model=None, chidx=None, cvidx=None):
+        """Return skew quadrupoles strengths."""
+        if model is None:
+            model = self.model
+        if chidx is None:
+            chidx = self.ch_idx
+        if cvidx is None:
+            cvidx = self.cv_idx
+        hkick = []
+        for mag in chidx:
+            hkick_seg = []
+            for seg in mag:
+                hkick_seg.append(model[seg].hkick_polynom)
+            hkick.append(hkick_seg)
+        vkick = []
+        for mag in cvidx:
+            vkick_seg = []
+            for seg in mag:
+                vkick_seg.append(model[seg].vkick_polynom)
+            vkick.append(vkick_seg)
+        hkick = np.array(hkick)
+        vkick = np.array(vkick)
+        return np.vstack((hkick, vkick))
+
+    def set_kicks(self, model=None, corridx=None, kicks=None):
+        """Set skew quadrupoles strengths in the model."""
+        if model is None:
+            model = self.model
+        if corridx is None:
+            corridx = self.corr_idx
+        if kicks is None:
+            raise Exception('Missing Kicks values')
+        newmod = _dcopy(model)
+        for idx_mag, mag in enumerate(corridx):
+            for idx_seg, seg in enumerate(mag):
+                kick = kicks[idx_mag][idx_seg]
+                if idx_mag < self._nch:
+                    if np.abs(kick) > OrbitCorr.MAX_HKICK:
+                        kick = np.sign(kick) * OrbitCorr.MAX_HKICK
+                    newmod[seg].hkick_polynom = kick
+                else:
+                    if np.abs(kick) > OrbitCorr.MAX_VKICK:
+                        kick = np.sign(kick) * OrbitCorr.MAX_VKICK
+                    newmod[seg].vkick_polynom = kick
+        return newmod
+
+    @staticmethod
+    def get_figm(res):
+        """Calculate figure of merit from residue vector."""
+        return np.sqrt(np.sum(res*res)/res.size)
+
+    def orbit_corr(self,
+                   model,
```

Comment:
I think model should be optional here. Just like in tune or chrom corrections, the default should be `self.model`